File: src/LeetCode/PalindromeNumber.cpp
#include <iostream>

using namespace std;

bool isPalindrome(int x) {
    string temp = to_string(x);
    int start = 0;
    int end = temp.length() - 1;
    while(start<end){
        if(temp[start]!=temp[end]){
            return false;
        }
        start++;
        end--;
    }
    return true;
}


int main(){
    int x;
    cin>>x;
    cout<<isPalindrome(x);

}

File: src/LeetCode/PalindromeNumber.java
package leetcode;

import java.util.Scanner;

public class PalindromeNumber {

    public static boolean isPalindrome(int x) {
//        if(x<0){
//            return false;
//        }
        String temp = String.valueOf(x);
        int start = 0;
        int end = temp.length()-1;
        boolean res = true;
        while(start<end){
            if(temp.charAt(start)!=temp.charAt(end)){
                res = false;
                return res;
            }
            start++;
            end--;
        }

        return res;
    }

    public static void main(String[] args){
        Scanner scanner = new Scanner(System.in);
        int x = scanner.nextInt();
        System.out.println(isPalindrome(x));

    }
}
